controller/api/destination/watcher/endpoints_watcher.go

```diff
@@ -483,6 +603,87 @@ func metricLabels(endpoints *corev1.Endpoints) map[string]string {
 	return labels
 }
 
+func (pp *portPublisher) endpointSliceToAddresses(es *dv1beta1.EndpointSlice) AddressSet {
+	addresses := make(map[ID]Address)
+
+	resolvedPort := pp.resolveESTargetPort(es.Ports)
+	for _, endpoint := range es.Endpoints {
+		if pp.hostname != "" && pp.hostname != *endpoint.Hostname {
+			continue
+		}
+
+		// Without this we get `Add:` twice, once when !ready
+		// and once when ready
+		if !*endpoint.Conditions.Ready {
+			continue
+		}
+
+		// Every endpoint has addresses that correspond to the EndpointSlice addressType field
+		// Consumers must handle different types of addresses
+		// An "addresses" set must contain at least 1 but no more than 100 addresses
+		if endpoint.TargetRef == nil {
+			serviceName := es.Labels[dv1beta1.LabelServiceName]
+			for _, addr := range endpoint.Addresses {
+				if isIPv6Address(addr) {
```

Comment:
Hm, I think this might cause some trouble here and there. So currently an EndpointsSlice has an address type defined such as: 

```
const (
	// AddressTypeIP represents an IP Address.
	// This address type has been deprecated and has been replaced by the IPv4
	// and IPv6 adddress types. New resources with this address type will be
	// considered invalid. This will be fully removed in 1.18.
	// +deprecated
	AddressTypeIP = AddressType("IP")
	// AddressTypeIPv4 represents an IPv4 Address.
	AddressTypeIPv4 = AddressType(v1.IPv4Protocol)
	// AddressTypeIPv6 represents an IPv6 Address.
	AddressTypeIPv6 = AddressType(v1.IPv6Protocol)
	// AddressTypeFQDN represents a FQDN.
	AddressTypeFQDN = AddressType("FQDN")
)
```

This means that the address can actually be a domain name like `google.com`. In this case this function will return false: 

```
func isIPv6Address(address string) bool {
	return strings.Count(address, ":") >= 2
}
```

Not sure what happens next but I think it is not what we want. Nevertheless the fact that there is an address type that can be a concrete netwrok address (IPv4 or IPv6) OR a domain name is a great feature that I think will solve some of the headscratchers we are having here at the moment